forge-ai/src/main/java/forge/ai/ability/PumpAi.java

```diff
@@ -104,6 +97,12 @@ protected boolean checkPhaseRestrictions(final Player ai, final SpellAbility sa,
                 return false;
             }
         }
+        if (sa.hasParam("SwitchPT")) {
```

Comment:
one of these blocks is redundant

Reply:
one is phase restrictions, the other is ApiLogic

Reply:
yes and both will be called anyway
it's just a helper for code structure...

Reply:
it is just that the default implementations for other Pump should be ignored for this